collector_test.go

```diff
@@ -9,6 +9,72 @@ import (
 	"github.com/prometheus/snmp_exporter/config"
 )
 
+func TestPduToSample(t *testing.T) {
+	cases := []struct {
+		pdu            *gosnmp.SnmpPDU
+		indexOids      []int
+		metric         *config.Metric
+		oidToPdu       map[string]gosnmp.SnmpPDU
+		expectedResult string
+	}{
+		{
+			pdu: &gosnmp.SnmpPDU{
+				Name:  "1.3.6.1.2.1.31.1.1.1.11.629",
+				Type:  0x02,
+				Value: 2,
+			},
+			indexOids: []int{1},
+			metric: &config.Metric{
+				Name: "test_metric",
+				Oid:  "1.3.6.1.2.1.31.1.1.1.11.629",
+				Type: "counter",
+				Help: "Help string",
+			},
+			oidToPdu:       make(map[string]gosnmp.SnmpPDU),
+			expectedResult: "Desc{fqName: \"test_metric\", help: \"Help string\", constLabels: {}, variableLabels: []}",
+		},
+		{
+			pdu: &gosnmp.SnmpPDU{
+				Name:  "1.3.6.1.2.1.31.1.1.1.11.629",
+				Type:  0x02,
+				Value: 2,
+			},
+			indexOids: []int{1},
+			metric: &config.Metric{
+				Name: "test_metric",
+				Oid:  "1.3.6.1.2.1.31.1.1.1.11.629",
+				Type: "gauge",
+				Help: "Help string",
+			},
+			oidToPdu:       make(map[string]gosnmp.SnmpPDU),
+			expectedResult: "Desc{fqName: \"test_metric\", help: \"Help string\", constLabels: {}, variableLabels: []}",
+		},
+		{
+			pdu: &gosnmp.SnmpPDU{
+				Name:  "1.3.6.1.2.1.31.1.1.1.11.629",
+				Type:  0x02,
+				Value: -2,
+			},
+			indexOids: []int{1},
+			metric: &config.Metric{
+				Name: "test_metric",
+				Oid:  "1.3.6.1.2.1.31.1.1.1.11.629",
+				Help: "Help string",
+			},
+			oidToPdu:       make(map[string]gosnmp.SnmpPDU),
+			expectedResult: "Desc{fqName: \"test_metric\", help: \"Help string\", constLabels: {}, variableLabels: [test_metric]}",
+		},
+	}
+
+	for _, c := range cases {
+		res := pduToSample(c.indexOids, c.pdu, c.metric, c.oidToPdu)
+		if res.Desc().String() != c.expectedResult {
```

Comment:
You'd probably be better off using `Write` rather than `Desc` so the values can be checked.